src/thread_task.rs

```diff
@@ -935,15 +960,15 @@ pub unsafe extern "C" fn rav1d_worker_task(data: *mut c_void) -> *mut c_void {
                     prev_t = Some(t);
                     f.task_thread.task_cur_prev = prev_t.as_deref_mut().unwrap() as *mut Rav1dTask;
                 }
-                ttd.cur = (ttd.cur).wrapping_add(1);
+                ttd.cur.fetch_add(1, Ordering::Relaxed);
```

Comment:
```suggestion
                ttd.cur.store(cur, Ordering::Relaxed);
```

and if `fn park`, etc. pass and return `delayed_fg` by value so borrowck can reason about disjoint field borrows, I think this could just be `ttd.cur = AtomicU32::new(cur);`.